### PR TITLE
redis: Memorystore Redis samples

### PR DESCRIPTION
**Pull request by laljikanjareeya**
Fixes: #944

### CODE REVIEW

**Comment by laljikanjareeya** [comment]
@amanda-tarafa It's looks like CI configured project does not have `Redis` Permission, getting below Error Please take a look.
`
Error: System.AggregateException : One or more errors occurred. (Exit code: 7 Status(StatusCode=PermissionDenied, Detail="Google Cloud Memorystore for Redis API has not been used in project 1061780273775 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/redis.googleapis.com/overview?project=1061780273775 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.")
`

**Comment by amanda-tarafa** [comment]
API enabled and permissions granted.

File: apis/Redis/Redis.Samples.Tests/RadisFixture.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Xunit;
+
+[CollectionDefinition(nameof(RedisFixture))]
+public class RedisFixture : IDisposable, ICollectionFixture<RedisFixture>
+{
+    private readonly CreateInstanceSample _createSample;
```
Comment (amanda-tarafa) [inline apis/Redis/Redis.Samples.Tests/RadisFixture.cs:21]: nit: Maybe don't store these two, just create them when you are using them.

File: apis/Redis/Redis.Samples.Tests/UpdateInstanceTest.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Xunit;
+
+[Collection(nameof(RedisFixture))]
+public class UpdateInstanceTest
+{
+    private readonly RedisFixture _fixture;
+    private readonly UpdateInstanceSample _sample;
```
Comment (amanda-tarafa) [inline apis/Redis/Redis.Samples.Tests/UpdateInstanceTest.cs:21]: Maybe same, don't keep these, just create them when you need them. Particularly the one you are testing should be created inside the test.

Here and for the rest.

File: apis/Redis/Redis.Samples.Tests/CreateInstanceTest.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        _sample = new CreateInstanceSample();
+    }
+
+    [Fact]
+    public void CreateInstance()
+    {
+        try
+        {
+            //run the sample code.
+            _sample.CreateInstance(_fixture.ProjectId, _fixture.LocationId, _fixture.InstanceId);
```
Comment (amanda-tarafa) [inline apis/Redis/Redis.Samples.Tests/CreateInstanceTest.cs:36]: Hmmm, maybe just create a fresh one and delete it as part of the test. Or add a MarkForDeletion method in the fixture that you can pass an instance Ids that are stored in a list and deleted on dispose.

File: apis/Redis/Redis.Samples/DeleteInstance.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    /// <summary>
+    /// Deletes a specific Redis instance. Instance stops serving and data is
+    /// deleted.
+    /// </summary>
+    public void DeleteInstance(string projectId = "my-project",
+        string locationId = "us-east1", string instanceId = "my-instance")
+    {
+        // Create client
+        CloudRedisClient cloudRedisClient = CloudRedisClient.Create();
+        DeleteInstanceRequest request = new DeleteInstanceRequest
```
[the remark this answers is not in the document]
Comment (laljikanjareeya) [reply apis/Redis/Redis.Samples/DeleteInstance.cs]: Done!

File: apis/Redis/Redis.Samples.Tests/RadisFixture.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    private readonly DeleteInstanceSample _deleteSample;
+    public string ProjectId { get; }
+    public string LocationId { get; }
+    public string InstanceId { get; private set; } = $"csharp-{Guid.NewGuid().ToString().Substring(0, 20)}";
+
+    public RedisFixture()
+    {
+        ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
```
[the remark this answers is not in the document]
Comment (laljikanjareeya) [reply apis/Redis/Redis.Samples.Tests/RadisFixture.cs]: I just throw exception when project ID is null or empty.

**Comment by laljikanjareeya** [comment]
Thanks @amanda-tarafa for the review. I just addressed all your comments also added test case for Delete Instance please take a look.

**Comment by laljikanjareeya** [comment]
> These are still timing out.

@amanda-tarafa  I have already increase the timeout for test case, still it's timing out.

We are creating three instance on test setup:
1. Create Instance Test
2. Delete Instance Test
3. Update, Get and List Instance Test

If We can skip creating instance in Create and Delete instance test than setup will be faster.
We are already creating/Deleting instance using sample in `fixture`.

**Comment by amanda-tarafa** [comment]
Closing because of lack of bandwidth. See #944.

**Pull request closed without merging**